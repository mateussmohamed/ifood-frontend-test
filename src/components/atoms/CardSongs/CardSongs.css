.card-song {
  color: #d6d6d6;
  font-size: 12px;
  font-weight: 300;
}
