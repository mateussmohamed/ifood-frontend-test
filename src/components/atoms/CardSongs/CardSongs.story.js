import { storiesOf } from '@storybook/react';
import React from 'react';
import CardSongs from './index';

const stories = storiesOf('CardSongs', module);

stories.add('with song props', () => <CardSongs songs="18" />);
