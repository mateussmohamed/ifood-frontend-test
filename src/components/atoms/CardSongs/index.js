import React from 'react';
import PropTypes from 'prop-types';

import './CardSongs.css';

const CardSongs = ({ songs }) => (
  <span className="card-song">{songs} Songs</span>
);

CardSongs.propTypes = {
  songs: PropTypes.number.isRequired
};

export default CardSongs;
