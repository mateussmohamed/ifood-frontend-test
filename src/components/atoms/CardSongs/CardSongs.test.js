import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CardSongs from './index';

test('CardSongs match to snapshot', () => {
  const component = renderer.create(<CardSongs songs={234} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CardSongs contains 29 songs when passed songs prop', () => {
  const component = shallow(<CardSongs songs={29} />);
  const rendered = <span className="card-song">29 Songs</span>;
  expect(component.contains(rendered)).toBeTruthy();
});
