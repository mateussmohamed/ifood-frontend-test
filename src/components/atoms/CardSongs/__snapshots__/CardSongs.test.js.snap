// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSongs match to snapshot 1`] = `
<span
  className="card-song"
>
  234
   Songs
</span>
`;
